airsonic-main/src/main/webapp/WEB-INF/jsp/playlist.jsp

```diff
@@ -285,6 +286,16 @@
                     playlistMusicTable.one( "draw", function () {
                         onRearrange(playlistMusicTable.rows().indexes().toArray());
                     });
+                    playlistMusicTable.order([0, "asc"]);
+                }
+            });
+            playlistMusicTable.on("order", (e, settings, ordArr) => {
+                if (previousSortingOrder.col !== ordArr[0].col || previousSortingOrder.dir !== ordArr[0].dir) {
```

Comment:
this seems lacking on the secondary (and further) sort order checks.

if you order by artist desc and then do an additional sort order on album asc, it'll save on the first order, but fail on the second order (because the first order remains identical)

also why are the checks necessary? does it save a call to the backend every time a draw is done (which would come back via websocket notification and cause another redraw, and then another order event, thus causing a loop)?

maybe just save the entire array as previousSortingOrder and deep compare the arrays?

Reply:
that's why i check not only column used for sorting but also sorting order, to enable multiple sort operations on the same column.
the check was necessary because 'order' event is raised on every table load, and table is reloaded after save so we had an endless loop without it